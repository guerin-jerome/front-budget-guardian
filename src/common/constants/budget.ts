export enum EBudgetType {
  VARIABLE = "Dépense variable",
  FIXE = "Dépense fixe",
  SAVED = "Épargne",
}
