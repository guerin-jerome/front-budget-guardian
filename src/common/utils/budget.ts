import { BudgetType, BudgetTypeLibelle } from "@/entities/Budget";
import { replacePointByComma } from "./string";

export const buildLibelleTypeBudget = (type: BudgetType): BudgetTypeLibelle => {
  switch (type) {
    case BudgetType.FIXE:
      return BudgetTypeLibelle.FIXE;
    case BudgetType.VARIABLE:
      return BudgetTypeLibelle.VARIABLE;
    case BudgetType.SAVED:
      return BudgetTypeLibelle.SAVED;
  }
};

export const formatNumberToPrice = (number: number) =>
  replacePointByComma(`${number.toFixed(2)} â‚¬`);
