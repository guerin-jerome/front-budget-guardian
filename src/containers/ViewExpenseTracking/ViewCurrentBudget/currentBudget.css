.budget {
  margin-top: 20px;
  border: 1px solid #ddd;
  padding: 10px;
}

.budget h3 {
  font-size: 18px;
  color: #333;
  margin: 0;
}

.budget-info {
  margin-top: 10px;
  display: flex;
  justify-content: space-between;
  align-items: center;
}

.budget-info .budget-type {
  font-size: 14px;
  color: #999;
}

.budget-info .budget-value {
  font-size: 16px;
  font-weight: bold;
  color: #333;
}

.progress-bar {
  height: 10px;
  background-color: #eee;
  margin-top: 10px;
}

.progress {
  height: 100%;
  background-color: #0077c2;
}
