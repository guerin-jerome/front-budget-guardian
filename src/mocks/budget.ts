import {
  Budget,
  BudgetFixe,
  BudgetSaved,
  BudgetType,
  BudgetVariable,
} from "@/entities/Budget";

export const MOCK_BUDGET: Budget = {
  id: 1,
  name: "Loisirs",
  type: BudgetType.VARIABLE,
  spending: 5.6,
  base: 100,
};

export const MOCK_BUDGET_SAVED: Budget = {
  id: 2,
  name: "Placement financier",
  type: BudgetType.SAVED,
  base: 200,
};

export const MOCK_BUDGET_FULL_CONSUMMED: Budget = {
  id: 3,
  name: "Loyer",
  type: BudgetType.FIXE,
  spending: 560,
  base: 560,
};

export const MOCK_BUDGETS: Budget[] = [
  { ...MOCK_BUDGET },
  { ...MOCK_BUDGET_SAVED },
  { ...MOCK_BUDGET_FULL_CONSUMMED },
  {
    id: 4,
    name: "Courses",
    type: BudgetType.VARIABLE,
    spending: 50,
    base: 175,
  },
];

export const SAVED_BUDGET: BudgetSaved = {
  id: 1,
  type: BudgetType.SAVED,
  name: "Saved budget",
  base: 100,
};

export const FIXE_BUDGET: BudgetFixe = {
  id: 2,
  type: BudgetType.FIXE,
  name: "Fixe budget",
  base: 100,
  spending: 100,
};

export const VARIABLE_BUDGET: BudgetVariable = {
  id: 3,
  type: BudgetType.VARIABLE,
  name: "Variable budget",
  base: 100,
  spending: 5,
};
