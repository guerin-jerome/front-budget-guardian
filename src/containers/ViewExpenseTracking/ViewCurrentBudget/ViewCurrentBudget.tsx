import { useCurrentBudget } from "../../../hooks/useCurrentBudget";
import { TBudget } from "../../../interfaces/TBudget";
import "./currentBudget.css";

export const ViewCurrentBudget = (budget: TBudget) => {
  const { name, type } = budget;
  const { libelleCurrentValue, percentConsummed } = useCurrentBudget(budget);

  return (
    <div className="budget">
      <h3>{name}</h3>
      <div className="budget-info">
        <span className="budget-type">{type}</span>
        <span className="budget-value">{libelleCurrentValue}</span>
      </div>
      <div className="progress-bar">
        <div className="progress" style={{ width: percentConsummed }} />
      </div>
    </div>
  );
};
