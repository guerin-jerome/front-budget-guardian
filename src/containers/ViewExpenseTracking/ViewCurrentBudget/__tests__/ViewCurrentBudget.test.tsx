import { render, screen } from "@testing-library/react";
import { describe, it, expect } from "vitest";
import { ViewCurrentBudget } from "../ViewCurrentBudget";
import { MOCK_BUDGET, MOCK_BUDGET_SAVED } from "../../../../mocks/budget.mock";
import { EBudgetType } from "../../../../common/constants/budget";

describe("<ViewCurrentBudget />", () => {
  it("matchs snapshot", () => {
    const { container } = render(<ViewCurrentBudget {...MOCK_BUDGET} />);
    expect(container).toMatchSnapshot();
  });

  it("renders well - budget variable", () => {
    render(<ViewCurrentBudget {...MOCK_BUDGET} />);
    const budgetName = screen.getByRole("heading", { name: "Loisirs" });
    const budgetType = screen.getByText(EBudgetType.VARIABLE);
    const budgetCurrentValue = screen.getByText("5,6€/100€");
    expect(budgetName).toBeDefined();
    expect(budgetType).toBeDefined();
    expect(budgetCurrentValue).toBeDefined();
  });

  it("renders well - budget saved", () => {
    render(<ViewCurrentBudget {...MOCK_BUDGET_SAVED} />);
    const budgetName = screen.getByRole("heading", {
      name: "Placement financier",
    });
    const budgetType = screen.getByText(EBudgetType.SAVED);
    const budgetCurrentValue = screen.getByText("200€");
    expect(budgetName).toBeDefined();
    expect(budgetType).toBeDefined();
    expect(budgetCurrentValue).toBeDefined();
  });
});
