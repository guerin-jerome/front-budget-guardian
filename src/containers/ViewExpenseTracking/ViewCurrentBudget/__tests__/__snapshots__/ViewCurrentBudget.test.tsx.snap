// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`<ViewCurrentBudget /> > matchs snapshot 1`] = `
<div>
  <div
    class="budget"
  >
    <h3>
      Loisirs
    </h3>
    <div
      class="budget-info"
    >
      <span
        class="budget-type"
      >
        Dépense variable
      </span>
      <span
        class="budget-value"
      >
        5,6€/100€
      </span>
    </div>
    <div
      class="progress-bar"
    >
      <div
        class="progress"
        style="width: 5%;"
      />
    </div>
  </div>
</div>
`;
